opslevel/resource_opslevel_check_repository_grep.go

```diff
@@ -113,6 +114,33 @@ func (r *CheckRepositoryGrepResource) Schema(ctx context.Context, req resource.S
 	}
 }
 
+func (r *CheckRepositoryGrepResource) UpgradeState(ctx context.Context) map[int64]resource.StateUpgrader {
+	predicateSchema := PredicateSchema()
+	predicateSchema.Optional = false
+	predicateSchema.Required = true
+
+	return map[int64]resource.StateUpgrader{
+		// State upgrade implementation from 0 (prior state version) to 1 (Schema.Version)
+		0: {
+			PriorSchema: &schema.Schema{
+				Attributes: CheckBaseAttributes(map[string]schema.Attribute{
+					"directory_search": schema.BoolAttribute{
+						Description: "Whether the check looks for the existence of a directory instead of a file.",
+						Required:    true,
+					},
+					"filepaths": schema.ListAttribute{
+						Description: "Restrict the search to certain file paths.",
+						Required:    true,
+						ElementType: types.StringType,
+					},
+					"file_contents_predicate": predicateSchema,
```

Comment:
```suggestion
					"file_contents_predicate": PredicateSchema(),
```
I'm seeing `PredicateSchema()` in most places and assume this is what's needed?

Reply:
@davidbloss actually this one uses a local variable up on lines 118.  This was copied directly from 1.2